heron/spi/src/java/com/twitter/heron/spi/common/Key.java

```diff
@@ -109,21 +109,22 @@
   INSTANCE_DISK             ("heron.resources.instance.disk", ByteAmount.fromBytes(1073741824)),
 
   //keys for checkpoint management
-  IS_STATEFUL                       ("heron.is.stateful", Boolean.FALSE),
-  IS_CLEAN_STATEFUL_CHECKPOINTS     ("heron.is.clean.stateful.checkpoints", Boolean.FALSE),
-  CKPTMGR_STATEFUL_STORAGE          ("heron.statefulstorage.type", Type.STRING),
-  STATEFUL_STORAGE_CONF             ("heron.statefulstorage.config", Type.MAP),
-  STATEFUL_STORAGE_CLASSPATH        ("heron.statefulstorage.classpath", Type.STRING),
+  IS_STATEFUL_ENABLED                      ("heron.is.stateful.enabled", Boolean.FALSE),
+  IS_CLEAN_STATEFUL_CHECKPOINTS            ("heron.is.clean.stateful.checkpoints", Boolean.FALSE),
+  STATEFUL_STORAGE_CLASSNAME               ("heron.statefulstorage.classname", Type.STRING),
+  STATEFUL_STORAGE_CONF                    ("heron.statefulstorage.config", Type.MAP),
+  STATEFUL_STORAGE_CUSTOM_CLASSPATH        ("heron.statefulstorage.custom.classpath", Type.STRING),
```

Comment:
`STATEFUL_STORAGES_CLASSPATH` is telling the jvm where are the stateful storage implementations (LocalFileSystemStorage, HDFSStorage).

`STATEFUL_STORAGES_CURTOM_CLASS` is telling the jvm where are additional classes needed by the stateful storage implementation. E.g., in HDFSStorage, we need to specify `  heron.statefulstorage.classpath:        "$(hadoop --config /path/to/configs classpath)"` to make the hdfs client working.